src/Stryker.Core/Stryker.Core/Initialisation/Buildalyzer/IAnalyzerResultExtensions.cs

```diff
@@ -103,6 +103,35 @@ public static string GetSymbolFileName(this IAnalyzerResult analyzerResult)
             return Path.ChangeExtension(analyzerResult.GetAssemblyName(), ".pdb");
         }
 
+        public static IEnumerable<ISourceGenerator> GetSourceGenerators(this IAnalyzerResult analyzerResult, ILogger logger = null)
+        {
+            var generators = new List<ISourceGenerator>();
+            foreach (var analyzer in analyzerResult.AnalyzerReferences)
+            {
+                try
+                {
+                    var assembly = System.Reflection.Assembly.LoadFile(analyzer);
+                    foreach (var type in assembly.ExportedTypes)
+                    {
+                        if (type.GetInterface("ISourceGenerator") != null)
+                        {
+                            var generator = type.GetConstructor(Type.EmptyTypes).Invoke(null);
+                            generators.Add(generator as ISourceGenerator);
+                        }
+                    }
+                }
+                catch (Exception e)
```

Comment:
Any chance to catch something more specific here?

Reply:
The most common exception is System.IO.FileNotFoundException - the issue reported by Cyrille above - other possibilities include, but are not limited to: TypeLoadException, FileLoadException, BadImageFormatException, ReflectionTypeLoadException. Considering the warning message is the same in each case, I don't believe there is any utility in singling out specific exceptions. @psfinaki

Reply:
I see, makes sense. Thanks for the detailed response, resolving this

Reply:
thanks